modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java

```diff
@@ -226,6 +272,212 @@ public void testDeleteDocument() {
+		throws IOException {
+
+		String imageFileName = "image.png";
+
+		_documentHref = ApioClientBuilder.given(
+		).basicAuth(
+			"test@liferay.com", "test"
+		).header(
+			"Accept", "application/hal+json"
+		).header(
+			"Content-Type", "multipart/form-data"
+		).multipart(
+			"binaryFile", FileTestUtil.getFile(imageFileName, getClass())
+		).when(
+		).post(
+			_documentsHref
+		).then(
+		).extract(
+		).path(
+			"_links.self.href"
+		);
+
+		Awaitility.reset();
+		Awaitility.setDefaultPollInterval(100, TimeUnit.MILLISECONDS);
+
+		Awaitility.await(
+		).atMost(
+			2, TimeUnit.SECONDS
+		).until(
+			() -> {
+				Object adaptiveMedia = ApioClientBuilder.given(
+				).basicAuth(
+					"test@liferay.com", "test"
+				).header(
+					"Accept", "application/hal+json"
+				).when(
+				).get(
+					_documentsHref
+				).then(
+				).extract(
+				).path(
+					"_embedded.'Liferay:Document'.find {it.title == '" +
+						imageFileName + "'}._embedded.adaptedMedia._embedded"
+				);
+
+				return adaptiveMedia != null;
+			}
+		);
+
+		String adaptiveMedia1ContentHref = ApioClientBuilder.given(
+		).basicAuth(
+			"test@liferay.com", "test"
+		).header(
+			"Accept", "application/hal+json"
+		).when(
+		).get(
+			_documentsHref
+		).then(
+		).extract(
+		).path(
+			"_embedded.'Liferay:Document'.find {it.title == '" + imageFileName +
+				"'}._embedded.adaptedMedia._embedded.find {it.resolutionName " +
+					"== 'Preview-1000x0'}.contentUrl"
+		);
+
+		URL adaptiveMedia1ContentURL = new URL(adaptiveMedia1ContentHref);
+
+		byte[] contentBytes1 = IOUtils.toByteArray(
+			adaptiveMedia1ContentURL.openStream());
+
+		Assert.assertTrue(contentBytes1.length > 0);
+
+		String adaptiveMedia2ContentHref = ApioClientBuilder.given(
+		).basicAuth(
+			"test@liferay.com", "test"
+		).header(
+			"Accept", "application/hal+json"
+		).when(
+		).get(
+			_documentsHref
+		).then(
+		).extract(
+		).path(
+			"_embedded.'Liferay:Document'.find {it.title == '" + imageFileName +
+				"'}._embedded.adaptedMedia._embedded.find {it.resolutionName " +
+					"== 'Thumbnail-300x300'}.contentUrl"
+		);
+
+		URL adaptiveMedia2ContentURL = new URL(adaptiveMedia2ContentHref);
+
+		byte[] contentBytes2 = IOUtils.toByteArray(
+			adaptiveMedia2ContentURL.openStream());
+
+		Assert.assertTrue(contentBytes2.length > 0);
+	}
+
+	@Test
+	public void testGetAdaptiveMediaFromImageInDocumentRepository() {
+		String imageFileName = "image.png";
+
+		_documentHref = ApioClientBuilder.given(
+		).basicAuth(
+			"test@liferay.com", "test"
+		).header(
+			"Accept", "application/hal+json"
+		).header(
+			"Content-Type", "multipart/form-data"
+		).multipart(
+			"binaryFile", FileTestUtil.getFile(imageFileName, getClass())
+		).when(
+		).post(
+			_documentsHref
+		).then(
+		).extract(
+		).path(
+			"_links.self.href"
+		);
+
+		Awaitility.reset();
+		Awaitility.setDefaultPollInterval(100, TimeUnit.MILLISECONDS);
```

Comment:
It is not possible to use awaitility default config?

Reply:
Yes, we can go with the default config too